subprojects/core/src/integTest/groovy/org/gradle/api/tasks/MissingTaskDependenciesIntegrationTest.groovy

```diff
@@ -282,6 +282,37 @@ class MissingTaskDependenciesIntegrationTest extends AbstractIntegrationSpec {
         executed(":producer", ":consumer")
     }
 
+    def "emits a deprecation warning when an input file collection can't be resolved"() {
+        buildFile """
+            task "broken" {
+                inputs.files(5).withPropertyName("invalidInputFileCollection")
+
+                doLast {
+                    println "success"
+                }
+            }
+        """
+        when:
+        executer.expectDocumentedDeprecationWarning("Consider using Task.dependsOn instead of an input file collection. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. Execution optimizations are disabled due to the failed validation. See https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.")
+        run "broken"
+        then:
+        executedAndNotSkipped ":broken"
+        outputContains("""
+            Validation failed for task ':broken', disabling optimizations:
```

Comment:
@lptr  Does this invalidate our conversation yesterday about 7.0 having no “disabled optimizations”? 

/cc @pioterj @facewindu

Reply:
Let's discuss this next week!